Fix for issue #66.  CreateInstance and remoting.

EMumau opened the pull request:
CreateInstance needs to PostInstance in order to suppress and error in the remoting code.  This also applies to issue #65.

This change adds support calling PostInstance from CreateInstance.  CreateInstance will still work without error (expect when remoting) without calling PostInstance.
Jumping, I tested remoting on a single box.  Would you please test this solution across boxes?
I also ran remoting with ei, di, gi, and mi.  Those all seem to work correctly when remoting without modification.

JumpingYang001 commented:
Nice fix! I will verify it on all platforms after build come out, might be next Monday, thank you!

JumpingYang001 commented:
BTW, what's difference between handle_post_result and handle_return? do we need to update other operations functions too?

JumpingYang001 reviewed (approved):
LGTM

EMumau commented:
handle_post_result only handles PostResult.

handle_return handles PostInstance and PostResult in a do while loop until (PostResult)


Sent from Outlook<http://aka.ms/weboutlook>


________________________________
From: Jumping Yang[Wicresoft] <notifications@github.com>
Sent: Friday, November 17, 2017 1:38 PM
To: Microsoft/omi-script-provider
Cc: Eric Mumau; Author
Subject: Re: [Microsoft/omi-script-provider] Fix for issue #66. CreateInstance and remoting. (#77)


BTW, what's difference between handle_post_result and handle_return? do we need to update other operations functions too?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fomi-script-provider%2Fpull%2F77%23issuecomment-345372587&data=02%7C01%7Cermumau%40microsoft.com%7C5280a7ebb9924ee7622908d52e0392f1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636465515240662023&sdata=2wVTu8USjvyDNZAI%2FTxJrcr2afyhlXeXXHkd%2FSRH7a8%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAUFs-cIqgnTDVFIergAgqc-Zh7_1G6uxks5s3fzigaJpZM4Qin7I&data=02%7C01%7Cermumau%40microsoft.com%7C5280a7ebb9924ee7622908d52e0392f1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636465515240662023&sdata=k4Yaug1aa41lJZEiAb5oKWZ1MjKlT49eFwDKlbaqtJI%3D&reserved=0>.

Commit:
Fix for issue #66.  CreateInstance needs to PostInstance in order to suppress and error in the remoting code.  This also applies to issue #65.
This change adds support calling PostInstance from CreateInstance.  CreateInstance will still work without error (expect when remoting) without calling PostInstance.
Jumping, I tested remoting on a single box.  Would you please test this solution across boxes?
I also ran remoting with ei, di, gi, and mi.  Those all seem to work correctly when remoting without modification.
